### PR TITLE
[SPARK-22178] [SQL] Refresh Persistent Views by REFRESH TABLE Command

### PR DESCRIPTION
**Pull request by gatorsmile**
## What changes were proposed in this pull request?
The underlying tables of persistent views are not refreshed when users issue the REFRESH TABLE command against the persistent views.

## How was this patch tested?
Added a test case

### CODE REVIEW

**Comment by gatorsmile** [comment]
cc @cloud-fan

File: sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala [hunk 1]
```diff
@@ -474,13 +474,20 @@ class CatalogImpl(sparkSession: SparkSession) extends Catalog {
    */
   override def refreshTable(tableName: String): Unit = {
     val tableIdent = sparkSession.sessionState.sqlParser.parseTableIdentifier(tableName)
-    // Temp tables: refresh (or invalidate) any metadata/data cached in the plan recursively.
-    // Non-temp tables: refresh the metadata cache.
-    sessionCatalog.refreshTable(tableIdent)
-
+    val tableMetadata = sessionCatalog.getTempViewOrPermanentTableMetadata(tableIdent)
     // If this table is cached as an InMemoryRelation, drop the original
```
Comment (viirya) [inline sql/core/src/main/scala/org/apache/spark/sql/internal/CatalogImpl.scala]: This comment should be moved to line 491.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetadataCacheSuite.scala [hunk 2]
```diff
@@ -31,14 +31,22 @@ import org.apache.spark.sql.test.SQLTestUtils
 class HiveMetadataCacheSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
 
   test("SPARK-16337 temporary view refresh") {
-    withTempView("view_refresh") {
+    checkRefreshView(isTemp = true)
+  }
+
+  test("view refresh") {
```
Comment (jiangxb1987) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetadataCacheSuite.scala]: We didn't cover the persistent view case for refresh, that's why the bug happens...

**Review by jiangxb1987 — APPROVED** [review]
LGTM

**Comment by viirya** [comment]
LGTM except for one minor comment.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetadataCacheSuite.scala (hunk 2 above)
Comment (gatorsmile) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveMetadataCacheSuite.scala]: Yes. We need to ask contributors for adding more test cases when reviewing the PR.

**Comment by gatorsmile** [comment]
Thanks! Merged to master/2.2

**Pull request closed without merging**